src/legacy/ui/public/visualize/loader/pipeline_helpers/utilities.ts

```diff
@@ -17,148 +17,13 @@
  * under the License.
  */
 
-import { i18n } from '@kbn/i18n';
-import { identity } from 'lodash';
-import { AggConfig } from 'ui/agg_types';
 import { npStart } from 'ui/new_platform';
-import { SerializedFieldFormat } from 'src/plugins/expressions/public';
 import { fieldFormats } from '../../../../../../plugins/data/public';
-import { Vis } from '../../../../../core_plugins/visualizations/public';
+import { SerializedFieldFormat } from '../../../../../../plugins/expressions/common/types';
 
-import { tabifyGetColumns } from '../../../agg_response/tabify/_get_columns';
-import { DateRangeKey, convertDateRangeToString } from '../../../agg_types/buckets/date_range';
-import { IpRangeKey, convertIPRangeToString } from '../../../agg_types/buckets/ip_range';
-
-interface TermsFieldFormatParams {
-  otherBucketLabel: string;
-  missingBucketLabel: string;
-  id: string;
-}
-
-function isTermsFieldFormat(
-  serializedFieldFormat: SerializedFieldFormat
-): serializedFieldFormat is SerializedFieldFormat<TermsFieldFormatParams> {
-  return serializedFieldFormat.id === 'terms';
-}
-
-const getConfig = (key: string, defaultOverride?: any): any =>
-  npStart.core.uiSettings.get(key, defaultOverride);
-const DefaultFieldFormat = fieldFormats.FieldFormat.from(identity);
-
-const getFieldFormat = (
-  id?: fieldFormats.IFieldFormatId,
-  params: object = {}
-): fieldFormats.FieldFormat => {
-  const fieldFormatsService = npStart.plugins.data.fieldFormats;
-
-  if (id) {
-    const Format = fieldFormatsService.getType(id);
-
-    if (Format) {
-      return new Format(params, getConfig);
-    }
-  }
-
-  return new DefaultFieldFormat();
+const createFormat = fieldFormats.serializeFieldFormat;
+const getFormat = (mapping?: SerializedFieldFormat) => {
+  return npStart.plugins.data.fieldFormats.deserialize(mapping as any);
```

Comment:
You may be able to get rid of `as any` with `!`, or by doing `if (mapping)`

```suggestion
  return mapping ? npStart.plugins.data.fieldFormats.deserialize(mapping) : undefined;
```
...or:
```suggestion
  return npStart.plugins.data.fieldFormats.deserialize(mapping!);
```